### DISCUSSION TITLE
same implementation for `min_maximal_matching` and `maximal_matching`?

### DISCUSSION THREAD

**Discussion by Schefflera-Arboricola** [networkx/networkx#7120] [Algorithms]
Currently, the function `min_maximal_matching` in [/algorithms/approximation/matching.py](https://github.com/networkx/networkx/blob/main/networkx/algorithms/approximation/matching.py#L17) returns the result of `nx.maximal_matching(G)`, which is also an independent function in [/algorithms/matching.py](https://github.com/networkx/networkx/blob/main/networkx/algorithms/matching.py#L21). Why do we have two functions implementing the same thing? is this a WIP? I couldn't find any PRs having the keyword `min_maximal_matching`. Also, the documentation of both functions doesn't seem aligned. I'd like to work on this issue, if this is an issue.

Thank you :)

**Comment by rossbar** [comment]
Linking #5435 here which has a similar theme.
